In [2]:
import pandas as pd

In [3]:
df = pd.read_csv('/Users/dipaksairamesh/Desktop/Projects/Python Projects/BCG - GenAI/Company_Financial_Data.csv')

In [4]:
df

Fiscal Year  Total Revenue  Net Income  Total Assets  Total Liabilities  \
0         2021         168088       61271        333779             191791   
1         2022         198270       72738        364840             198298   
2         2023         211915       72361        411976             205753   
3         2021          53823        5519         62131              30548   
4         2022          81462       12556         82338              36440   
5         2023          96773       14974        106618              43009   
6         2021         365817       94680        351002             287912   
7         2022         394328       99803        352755             302083   
8         2023         383285       96995        352583             290437   

   Cash Flow from Operating Activities    Company  
0                                76740  Microsoft  
1                                89035  Microsoft  
2                                87582  Microsoft  
3                                11497      Tesla  
4                                14724      Tesla  
5                                13256      Tesla  
6                               104038      Apple  
7                               122151      Apple  
8                               110543      Apple

In [5]:
df['Revenue Growth (%)'] = df.groupby(['Company'])['Total Revenue'].pct_change() * 100

In [6]:
df

Fiscal Year  Total Revenue  Net Income  Total Assets  Total Liabilities  \
0         2021         168088       61271        333779             191791   
1         2022         198270       72738        364840             198298   
2         2023         211915       72361        411976             205753   
3         2021          53823        5519         62131              30548   
4         2022          81462       12556         82338              36440   
5         2023          96773       14974        106618              43009   
6         2021         365817       94680        351002             287912   
7         2022         394328       99803        352755             302083   
8         2023         383285       96995        352583             290437   

   Cash Flow from Operating Activities    Company  Revenue Growth (%)  
0                                76740  Microsoft                 NaN  
1                                89035  Microsoft           17.956071  
2                                87582  Microsoft            6.882030  
3                                11497      Tesla                 NaN  
4                                14724      Tesla           51.351653  
5                                13256      Tesla           18.795267  
6                               104038      Apple                 NaN  
7                               122151      Apple            7.793788  
8                               110543      Apple           -2.800461

In [7]:
df['Net Income Growth (%)'] = df.groupby(['Company'])['Net Income'].pct_change() * 100

In [8]:
df

Fiscal Year  Total Revenue  Net Income  Total Assets  Total Liabilities  \
0         2021         168088       61271        333779             191791   
1         2022         198270       72738        364840             198298   
2         2023         211915       72361        411976             205753   
3         2021          53823        5519         62131              30548   
4         2022          81462       12556         82338              36440   
5         2023          96773       14974        106618              43009   
6         2021         365817       94680        351002             287912   
7         2022         394328       99803        352755             302083   
8         2023         383285       96995        352583             290437   

   Cash Flow from Operating Activities    Company  Revenue Growth (%)  \
0                                76740  Microsoft                 NaN   
1                                89035  Microsoft           17.956071   
2                                87582  Microsoft            6.882030   
3                                11497      Tesla                 NaN   
4                                14724      Tesla           51.351653   
5                                13256      Tesla           18.795267   
6                               104038      Apple                 NaN   
7                               122151      Apple            7.793788   
8                               110543      Apple           -2.800461   

   Net Income Growth (%)  
0                    NaN  
1              18.715216  
2              -0.518299  
3                    NaN  
4             127.504983  
5              19.257725  
6                    NaN  
7               5.410858  
8              -2.813543

In [9]:
df.fillna(0, inplace=True)

In [10]:
df

Fiscal Year  Total Revenue  Net Income  Total Assets  Total Liabilities  \
0         2021         168088       61271        333779             191791   
1         2022         198270       72738        364840             198298   
2         2023         211915       72361        411976             205753   
3         2021          53823        5519         62131              30548   
4         2022          81462       12556         82338              36440   
5         2023          96773       14974        106618              43009   
6         2021         365817       94680        351002             287912   
7         2022         394328       99803        352755             302083   
8         2023         383285       96995        352583             290437   

   Cash Flow from Operating Activities    Company  Revenue Growth (%)  \
0                                76740  Microsoft            0.000000   
1                                89035  Microsoft           17.956071   
2                                87582  Microsoft            6.882030   
3                                11497      Tesla            0.000000   
4                                14724      Tesla           51.351653   
5                                13256      Tesla           18.795267   
6                               104038      Apple            0.000000   
7                               122151      Apple            7.793788   
8                               110543      Apple           -2.800461   

   Net Income Growth (%)  
0               0.000000  
1              18.715216  
2              -0.518299  
3               0.000000  
4             127.504983  
5              19.257725  
6               0.000000  
7               5.410858  
8              -2.813543

In [11]:
df.to_csv('Analysis-Revenue&NetIncome.csv')

In [12]:
df['Assets Growth (%)'] = df.groupby(['Company'])['Total Assets'].pct_change() * 100
df['Liabilities Growth (%)'] = df.groupby(['Company'])['Total Liabilities'].pct_change() * 100
df['Cash Flow from Operating Activities Growth (%)'] = df.groupby(['Company'])['Cash Flow from Operating Activities'].pct_change() * 100

In [13]:
df

Fiscal Year  Total Revenue  Net Income  Total Assets  Total Liabilities  \
0         2021         168088       61271        333779             191791   
1         2022         198270       72738        364840             198298   
2         2023         211915       72361        411976             205753   
3         2021          53823        5519         62131              30548   
4         2022          81462       12556         82338              36440   
5         2023          96773       14974        106618              43009   
6         2021         365817       94680        351002             287912   
7         2022         394328       99803        352755             302083   
8         2023         383285       96995        352583             290437   

   Cash Flow from Operating Activities    Company  Revenue Growth (%)  \
0                                76740  Microsoft            0.000000   
1                                89035  Microsoft           17.956071   
2                                87582  Microsoft            6.882030   
3                                11497      Tesla            0.000000   
4                                14724      Tesla           51.351653   
5                                13256      Tesla           18.795267   
6                               104038      Apple            0.000000   
7                               122151      Apple            7.793788   
8                               110543      Apple           -2.800461   

   Net Income Growth (%)  Assets Growth (%)  Liabilities Growth (%)  \
0               0.000000                NaN                     NaN   
1              18.715216           9.305858                3.392756   
2              -0.518299          12.919636                3.759493   
3               0.000000                NaN                     NaN   
4             127.504983          32.523217               19.287678   
5              19.257725          29.488207               18.026894   
6               0.000000                NaN                     NaN   
7               5.410858           0.499427                4.921990   
8              -2.813543          -0.048759               -3.855232   

   Cash Flow from Operating Activities Growth (%)  
0                                             NaN  
1                                       16.021631  
2                                       -1.631942  
3                                             NaN  
4                                       28.068192  
5                                       -9.970117  
6                                             NaN  
7                                       17.409985  
8                                       -9.502992

In [14]:
df.fillna(0, inplace=True)

In [15]:
df

Fiscal Year  Total Revenue  Net Income  Total Assets  Total Liabilities  \
0         2021         168088       61271        333779             191791   
1         2022         198270       72738        364840             198298   
2         2023         211915       72361        411976             205753   
3         2021          53823        5519         62131              30548   
4         2022          81462       12556         82338              36440   
5         2023          96773       14974        106618              43009   
6         2021         365817       94680        351002             287912   
7         2022         394328       99803        352755             302083   
8         2023         383285       96995        352583             290437   

   Cash Flow from Operating Activities    Company  Revenue Growth (%)  \
0                                76740  Microsoft            0.000000   
1                                89035  Microsoft           17.956071   
2                                87582  Microsoft            6.882030   
3                                11497      Tesla            0.000000   
4                                14724      Tesla           51.351653   
5                                13256      Tesla           18.795267   
6                               104038      Apple            0.000000   
7                               122151      Apple            7.793788   
8                               110543      Apple           -2.800461   

   Net Income Growth (%)  Assets Growth (%)  Liabilities Growth (%)  \
0               0.000000           0.000000                0.000000   
1              18.715216           9.305858                3.392756   
2              -0.518299          12.919636                3.759493   
3               0.000000           0.000000                0.000000   
4             127.504983          32.523217               19.287678   
5              19.257725          29.488207               18.026894   
6               0.000000           0.000000                0.000000   
7               5.410858           0.499427                4.921990   
8              -2.813543          -0.048759               -3.855232   

   Cash Flow from Operating Activities Growth (%)  
0                                        0.000000  
1                                       16.021631  
2                                       -1.631942  
3                                        0.000000  
4                                       28.068192  
5                                       -9.970117  
6                                        0.000000  
7                                       17.409985  
8                                       -9.502992

In [16]:
df.to_csv('Analysis-Growth.csv')

In [17]:
summary = df.groupby('Company').agg({
    'Revenue Growth (%)': 'mean',
    'Net Income Growth (%)': 'mean',
    'Assets Growth (%)' : 'mean',
    'Liabilities Growth (%)' : 'mean',
    'Cash Flow from Operating Activities Growth (%)' :'mean'    
}).reset_index()

In [18]:
summary

Company  Revenue Growth (%)  Net Income Growth (%)  Assets Growth (%)  \
0      Apple            1.664442               0.865772           0.150223   
1  Microsoft            8.279367               6.065639           7.408498   
2      Tesla           23.382306              48.920903          20.670475   

   Liabilities Growth (%)  Cash Flow from Operating Activities Growth (%)  
0                0.355586                                        2.635664  
1                2.384083                                        4.796563  
2               12.438191                                        6.032692

In [19]:
summary.to_csv('Summary_Report_Data.csv')

### Summary of Financial Analysis Findings over 3 year fiscal period

- **Revenue Growth Trends:**
  - Apple showed a modest average revenue growth of 1.664442%.
  - Microsoft had a higher average revenue growth rate of 8.279367%.
  - Tesla had a substantial average revenue growth of 23.382306%.

- **Net Income Growth Trends:**
  - Apple's net income growth averaged 0.865772%.
  - Microsoft and Tesla also showed varying net income growth rates, 6.0656639% and 48.920903% respectively, indicating differing levels of profitability and cost management efficiency.

- **Assets and Liabilities Growth:**
  - Apple's assets grew on average by 0.150223%, while liabilities growth was minimal at 0.355586%, suggesting stable financial management.
  - Microsoft and Tesla revealed more significant fluctuations in asset and liability growth rates, reflecting different investment or financial strategies.

- **Cash Flow from Operating Activities:**
  - Growth rates varied significantly among companies, with some showing strong growth in operational cash flow, indicating robust operational health.

### Analysis Narratives

- **Revenue and Net Income Growth:**
  - The analysis indicates a general trend of growth in revenue across the companies, with Microsoft showing a particularly strong increase. This suggests effective market penetration and product acceptance.
  - Net income growth rates are less consistent, which could reflect varying cost structures and operational efficiencies. Apple's relatively stable net income growth suggests effective cost management despite modest revenue growth.

- **Asset Management:**
  - The asset growth rates are generally low, suggesting cautious capital management. Companies with minimal growth in liabilities alongside asset growth are likely improving their financial leverage and stability.

- **Operational Efficiency:**
  - Variations in cash flow from operating activities growth rates highlight differences in operational efficiency and cash management strategies among the companies. High growth in this area typically indicates more efficient operations and better cash management.